recipes/cfgfile/0.2.9.0/test_package/CMakeLists.txt

```
@@ -0,0 +1,26 @@
+cmake_minimum_required(VERSION 3.1)
+
+project(cfgfile.test)
+
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+
+conan_basic_setup(TARGETS)
+
+find_package(cfgfile REQUIRED)
+
+add_custom_target(gencfg
+   COMMAND ${CONAN_BIN_DIRS_CFGFILE}/cfgfile.generator${CMAKE_EXECUTABLE_SUFFIX} -i ${CMAKE_CURRENT_SOURCE_DIR}/session_cfg.conf -o ${CMAKE_CURRENT_BINARY_DIR}/cfg.hpp
+	)
+
+include_directories(${CMAKE_CURRENT_BINARY_DIR})
+
+add_executable(${PROJECT_NAME} example.cpp)
+
+add_dependencies(${PROJECT_NAME} gencfg)
```

1. There's no need to use the cmake variables CONAN_BIN_DIRS_CFGFILE} and CMAKE_EXECUTABLE_SUFFIX. CMAKE_EXECUTABLE_SUFFIX would be wrong anyway when cross building from e.g. Linux to Windows.
2. Use `add_custom_command` instead of `add_custom_target`.

See the docs for [`add_custom_target`](https://cmake.org/cmake/help/latest/command/add_custom_target.html): 
> The target has no output file and is always considered out of date even if the commands try to create a file with the name of the target. Use the add_custom_command() command to generate a file with dependencies. 

So this avoid re-running `cfgfile.generator` unnecessarily.
```suggestion
add_custom_command(
    OUTPUT "${CMAKE_CURRENT_BINARY_DIR}/cfg.hpp"
    COMMAND cfgfile.generator -i "${CMAKE_CURRENT_SOURCE_DIR}/session_cfg.conf" -o "${CMAKE_CURRENT_BINARY_DIR}/cfg.hpp"
    DEPENDS "${CMAKE_CURRENT_SOURCE_DIR}/session_cfg.conf"
)

include_directories(${CMAKE_CURRENT_BINARY_DIR})

add_executable(${PROJECT_NAME}
    example.cpp
    "${CMAKE_CURRENT_BINARY_DIR}/cfg.hpp"
)
```